File: atcoder.jp/contests/m-solutions2020/tasks/m_solutions2020_b/main.py
a, b, c, K = map(int, open(0).read().split())


for _ in range(K + 1):
    if a < b < c:
        print("Yes")
        exit()

    if a >= b:
        b *= 2
    elif b >= c:
        c *= 2


print("No")

File: atcoder.jp/contests/m-solutions2020/tasks/m_solutions2020_b/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	var a, b, c, K int
	fmt.Scanf("%d %d %d", &a, &b, &c)
	fmt.Scanf("%d", &K)

	for i := 0; i <= K; i++ {
		if a < b && b < c {
			fmt.Println("Yes")
			os.Exit(0)
		}

		if a >= b {
			b *= 2
		} else if b >= c {
			c *= 2
		}
	}

	fmt.Println("No")
}
